invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/Identifiers/PIDField/OptionalPIDField.js

```diff
@@ -0,0 +1,243 @@
+// This file is part of Invenio-RDM-Records
+// Copyright (C) 2020-2025 CERN.
+//
+// Invenio-RDM-Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import _debounce from "lodash/debounce";
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { FieldLabel } from "react-invenio-forms";
+import { Form } from "semantic-ui-react";
+import {
+  ManagedIdentifierCmp,
+  OptionalDOIoptions,
+  UnmanagedIdentifierCmp,
+} from "./components";
+import { getFieldErrors } from "./components/helpers";
+import { SET_DOI_NEEDED } from "../../../state/types";
+import { set } from "lodash";
+
+const PROVIDER_EXTERNAL = "external";
+const UPDATE_PID_DEBOUNCE_MS = 200;
+
+/**
+ * Render managed or unamanged PID fields and update
+ * Formik form on input changed.
+ * The field value has the following format:
+ * { 'doi': { identifier: '<value>', provider: '<value>', client: '<value>' } }
+ */
+class OptionalPIDFieldCmp extends Component {
+  constructor(props) {
+    super(props);
+
+    const { canBeManaged, canBeUnmanaged, record, field } = this.props;
+    this.canBeManagedAndUnmanaged = canBeManaged && canBeUnmanaged;
+    const value = field?.value;
+    const isInternalProvider = value?.provider !== PROVIDER_EXTERNAL;
+    const isDraft = record?.is_draft === true;
+    const hasIdentifier = value?.identifier;
+    const isManagedSelected =
+      isDraft && hasIdentifier && isInternalProvider ? true : undefined;
+
+    this.state = {
+      isManagedSelected: isManagedSelected,
+      isNoNeedSelected: undefined,
+    };
+  }
+
+  onExternalIdentifierChanged = (identifier) => {
+    const { form, fieldPath } = this.props;
+
+    const pid = {
+      identifier: identifier,
+      provider: PROVIDER_EXTERNAL,
+    };
+
+    this.debounced && this.debounced.cancel();
+    this.debounced = _debounce(() => {
+      form.setFieldValue(fieldPath, pid);
+    }, UPDATE_PID_DEBOUNCE_MS);
+    this.debounced();
+  };
+
+  render() {
+    const { isManagedSelected, isNoNeedSelected } = this.state;
+    const {
+      btnLabelDiscardPID,
+      btnLabelGetPID,
+      canBeManaged,
+      canBeUnmanaged,
+      form,
+      fieldPath,
+      fieldLabel,
+      isEditingPublishedRecord,
+      managedHelpText,
+      pidLabel,
+      pidIcon,
+      pidPlaceholder,
+      required,
+      unmanagedHelpText,
+      pidType,
+      field,
+      record,
+      optionalDOItransitions,
+      setNoINeedDOI,
+    } = this.props;
+
+    let { doiDefaultSelection } = this.props;
```

Comment:
Trivial: you can use `const` instead, and below
```suggestion
    const { doiDefaultSelection } = this.props;
```